### PR TITLE
Local docker development

### PR DESCRIPTION
**Pull request by benjaoming**
## Description
This PR adds a hopefully simple pain-free setup for backend developers to quickly deploy a local version of the frontend.

My own use-case: I don't have a good local Node/JS setup, I'm not a domain expert, and I will immediately get rabbit hole'd into figuring out what I'm doing. If someone provides me a Docker setup, I can quickly see what it's doing and will feel more at ease :)

Additional use-case: I understand that it might speed up development workflow if the frontend starts up very quickly, as might be the case for the "static" setup.

## Screenshots

N/A

## Changes

* Adds a Docker Compose setup + documentation in README
* Changes NOTHING

## Notes to reviewer

Maybe try on Mac?

## Related issues

Nay, onboarding

### CODE REVIEW

**Comment by richardolsson** [comment]
I think this PR was an experiment that didn't pan out entirely. Let me know if you want to reopen and fix the problems.

**Comment by benjaoming** [comment]
I can reopen and fix the linting error in index.tsx? Otherwise, it worked fine IRC.

**Comment by richardolsson** [comment]
> I can reopen and fix the linting error in index.tsx? Otherwise, it worked fine IRC.

Sure!

**Review by richardolsson — CHANGES_REQUESTED** [review]
I'm sorry I wasn't more clear in my previous comment but the reason why I thought of this as an experiment was because it was making little changes to code outside the scope of the PR. Before I'd be willing to merge this, we need to get rid of that unrelated code.

File: src/pages/index.tsx
Comment (benjaoming) [reply src/pages/index.tsx:99]: This is necessary because the frontend is built in a static way. Thus the environment is passed during `RUN NEXT_PUBLIC_APP_ENV=development yarn build`.

The static build was suggested by some of my fellow developers to speed up starting the frontend :innocent: 

But I can just remove it.. scope creeping is bad.. I just wanted a quick, dockerized shortcut.

**Comment by richardolsson** [comment]
You are misunderstanding me @benjaoming. Let's talk about this after lunch and give this another go.

**Comment by richardolsson** [comment]
Thanks for making those changes!

I tried running this, but I get errors from `docker compose` when it's reading the configuration from the YAML file. I'm guessing it's a version incompatibility. What version of docker and docker compose are you using? Here's mine:

```
app.zetkin.org % docker --version
Docker version 20.10.21, build baeda1f
app.zetkin.org % docker-compose --version
Docker Compose version v2.13.0
```

Not sure how to get the version of the "plugin" variant of docker compose that runs as `docker compose`, if that even makes sense.

**Comment by benjaoming** [comment]
Very strange!

```
➜ docker compose version  
Docker Compose version v2.25.0
```

However, I have a Docker Compose v2.16 which also doesn't work.

I understood from a Docker Compose issue that something got fixed in version `2.17.3`.

**Comment by richardolsson** [comment]
I tried this on my Mac, and it works fine, except for when the backend is also running on the same machine (which is one of the main use cases, so that's unfortunate).

I believe this is because the frontend container can't reach the backend container because there is no feasible network route from within the frontend container to the backend container. I believe on Linux the network host is the same for all containers and the host system, but that's not the case on Mac (or Windows, I believe).

Connecting to 127.0.0.1 from within the frontend container doesn't work (the way it does when running the frontend directly on the host machine) because that would be the container itself, not the host machine. And using a real hostname (e.g. api.dev.zetkin.org, which would be the normal way without docker) doesn't work because it will not resolve correctly to the host machine. Editing the hosts file in the frontend container doesn't work because that just resolves to 127.0.0.1 (the container). And editing the hosts file on the host machine also doesn't work because that is not where the hostname is resolved (that happens in the container).

In theory, this could be solved by figuring out the host machine IP from within the container, which will differ between operating systems and machines, so that adds another level of complexity entirely.

But I don't think we should spend more time on this now. If it works for you, and for other linux users, there's still value here. I'm just not sure whether we should merge and recommend this solution to users unless we also state very clearly that it's not expected to work on all systems, because that would only cause more confusion, not less.

**Comment by benjaoming** [comment]
> I believe this is because the frontend container can't reach the backend container because there is no feasible network route from within the frontend container to the backend container. 

Interesting! Would there be a quick way to supply the right IP or hostname as a parameter for either containers? (my question is badly phrased because I'm not sure that I understood the problem exactly)

Otherwise, I'll just make a quick note in the README that Mac users that are motivated to diagnose and propose a fix should feel welcome.

No, I don't think you should fix this before an actual Mac user comes and say that they need it :heart:

**Comment by benjaoming** [comment]
Also.... [insert unhappy Docker rant here]

File: next.config.js
Comment (richardolsson) [inline next.config.js:1]: Is there a reason for this change or did you add it accidentally to git? I prefer to keep the git log clean and only include relevant changes, because it makes git forensics (with `git log <file>` or `git blame` cleaner).

**Review by richardolsson — COMMENTED** [review]
Thanks for adding the information to the README. I found some unexpected extra whitespace and am curious before merging if that's intentional. See below.

File: next.config.js
Comment (benjaoming) [reply next.config.js]: ah yes, some mess from a previous change

**Review by richardolsson — APPROVED** [review]
Beautiful! I will wait for checks to finish and then merge if all is well.

**Pull request merged**
